source/languageDetection.py

```diff
@@ -0,0 +1,237 @@
+
+def detectScript(text):
+	"""splits a string if there are multiple scripts in it
+	@param text: the text string
+	@type string
+	@return: sequence of script commands and text
+	@rtype: list"""
+	unicodeSequence = []
+	outerScripts  = []
+	currentScript = getScriptCode(  text[0])
+	oldScript = currentScript
+	unicodeSequence.append(ScriptChangeCommand(currentScript)) 
+	beginIndex = 0
+	for index in xrange( len(text) ) :
+		currentScript = getScriptCode( text[index] ) 
+		# handling script property for paired punctuation: see UAX 24 section 5.1
+		if unicodedata.category(text[index] ) == "Ps":
+			outerScripts.append(oldScript ) 
+		elif unicodedata.category(text[index] ) == "Pe" and len(outerScripts) > 0:
+			currentScript = outerScripts.pop()
+
+		# don't change script for combining marks as per UAX 24 section 5.2
+		if unicodedata.category(text[index] ) in ["Mn" , "Mc", "Me"]: 
+			continue
+
+		# don't change scripts for Common, inherited and Unknown characters 
+		if currentScript in ["Common" , "Inherited", "Unknown"]: 
+			continue
+
+		if currentScript != oldScript:
+			newText = text[beginIndex:index] 
+			unicodeSequence.append( newText )
+			beginIndex= index
+			unicodeSequence.append(ScriptChangeCommand(currentScript)) 
+		oldScript = currentScript
+
+	unicodeSequence.append( text[beginIndex:] )
+	return unicodeSequence
+
+def detectLanguage(text, defaultLanguage =None):
+	"""splits a string if there are multiple languages in it. uses detectScript
+	@param text: the text string
+	@type text: string
+	@param defaultLanguage: The default language for NVDA 
+	@type preferredLanguage: string or None
+	@return: sequence of language commands and text
+	@rtype: list"""
+	sequenceWithLanguage= []
+	if config.conf["languageDetection"]["disableScriptDetection"]:  
+		sequenceWithLanguage.append( text)
+		return sequenceWithLanguage
+
+	tempSequence = detectScript(text)
+	scriptCode = ""
+
+	if defaultLanguage:
+		scriptIDForDefaultLanguage = getScriptIDFromLangID( defaultLanguage )
+	else:
+		scriptIDForDefaultLanguage = None
+
+	previousLanguageCode = ""
+	for index in xrange(len(tempSequence )):
+		item= tempSequence [index]
+		if isinstance(item,ScriptChangeCommand):
+			# check if default language for a script is available, if yes, add that language instead of language from priority list
+			if scriptIDForDefaultLanguage and (item.scriptCode in scriptIDForDefaultLanguage ):
+				if index == 0: continue # if it is first item and same as the default language, language code is already added.
+				languageCode = defaultLanguage 
+			else:
+				languageCode = getLangID( item.scriptCode  )  
+			#end if scriptIDForPreferredLanguage and (item.scriptCode == scriptIDForPreferredLanguage):
+
+			if languageCode:
+				if (languageCode == previousLanguageCode) or ( ( previousLanguageCode == "") and  (languageCode == defaultLanguage ) ): continue # if 2 scripts have same language, we don't need to add additional language. 
```

Comment:
please split this line up.

Reply:
I am unable to understand why do we need to split this line?

Reply:
For sighted developers much of the important information including the comment gets pushed off the screen. This makes it error prone to read the code. It's a fairly standard requirement to keep lines less than some maximum width, perhaps 80, 100, or 120 characters. See the [Pep 8 on the topic](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) which unfortunately doesn't go into why this is a good idea. My experience has been that its very easy to miss-read long lines, or not realise there is a comment on the end of it.